apps/frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,60 @@
+import { type ReactNode, useEffect, useRef } from "react";
+
+import crossIconSrc from "~/assets/images/icons/cross.svg";
+
+import styles from "./modal.module.css";
+
+type Properties = {
+	children: ReactNode;
+	onClose: () => void;
+	open: boolean;
+	title: string;
+};
+
+const Modal = ({ children, onClose, open, title }: Properties): JSX.Element => {
+	const dialogReference_ = useRef<HTMLDialogElement>(null);
+
```

Comment:
Create a separate custom hook to handle click outside useHandleClickOutside. It will accept ref and onClick as parameters and inside it you will need to add an event listener on document click. Then inside it compare if your click target node contains passed ref and target is not the ref then you should call onClick